tests/goal.test.sh

```
@@ -1475,6 +1482,65 @@ EOF
+{"pr":42,"sha":"abc123","phases":{"phase2_5":{"by_severity":{"blocker":1,"critical":0},"halted":false}}}
+EOF
+_bt23c_out="$(
+  gh() { case "$1 $2" in "pr view") printf 'def456' ;; esac; }
+  uberdev_goal_read_trust_signal "$_bt23c_v" 2>/dev/null
+)"
+assert_eq "$_bt23c_out" "stale" "BT23c.sha-mismatch-overrides-red-to-stale"
+
+# BT23d — gh outage reading headRefOid (rc!=0 / empty) → fail-SAFE: fall through
+# to the colour decision (green here), NOT a false stale (which would churn).
+_bt23d_out="$(
+  gh() { return 1; }
```

Shellcheck (0.11.0)

[info] This function is never invoked. Check usage (or ignored if invoked indirectly).

(SC2329)